scripts/init_devdax.sh

```diff
@@ -0,0 +1,129 @@
+    error_record_and_exit $? "ipmctl version "
+  fi
+
+  if [ $ipmctl_version -ne 2 ];then
+    error_record_and_exit $? "ipmctl version is less than 2,
+                     can't sub_dividing the region.
+                     It's better to recreate the region after upgrade impctl.
+                     ref issue: https://github.com/pmem/ndctl/issues/181"
+    exit $?
+  fi
+
+  ndctl list 2>/dev/null | error_record_and_exit $? "ndctl list "
+}
+
+function print_available_regions() {
+  echo "Available regions infomation: "
+  ndctl list --regions 2>/dev/null | \
+  jq '.[] | if .persistence_domain=="memory_controller" then select(.available_size>0) else empty end'
+  error_record_and_exit $? "ndctl list --regions | jq ..."
+}
+
+# Check input region name is exist and the namespace size is slower than available size.
+function check_input_region() {
+  region=${1}
+  size=${2}
+
+  # check input region is exists
+  ndctl list --regions 2>/dev/null | jq '.[].dev' | grep "$region"
+  error_record_and_exit $? "ndctl list --regions 2>/dev/null | jq '.[].dev' | grep -w '$region'"
+
+  # check input size is satisfy the available size
+  result=`ndctl list --regions 2>/dev/null | \
+          jq '.[] | select(.dev=='"\"$region\""') | if .available_size-'"$FSDAX_NAMESPACE_SIZE"' >= '"$size"' then "true" else "false" end'`
+  if [ $result == \""true\"" ];then
+    ret=0
+  else
+    ret=1
+  fi
+  error_record_and_exit $ret "ndctl list --regions 2>/dev/null | jq '.[] | select(.dev=='"\"$region\""') "
+}
+
+function check_namespace_type() {
+  type=${1}
+  echo "$NAMESPACE_TYPE_LISTS" |grep -w "$type" | \
+  error_record_and_exit $? 'echo "$NAMESPACE_TYPE_LISTS" |grep -w "$type'
+}
+
+
+check_available_tools
+print_available_regions
+
+read -p "input region name: " region_name
+read -p "input namespace size: " size
+read -p "input namespace type (fsdax | devdax | raw | sector):" namespace_type
+
+check_input_region "$region_name" "$size"
+check_namespace_type "$namespace_type"
+
+# Create the first mode
+ndctl create-namespace --region="$region_name" --size="$size" --mode="$namespace_type" |\
+error_record_and_exit $? "ndctl create-namespace ... create the $namespace_type namespace"
+echo "Success: "
+echo "create $namespace_type model with $size in $region_name"
+
+# Create the second mode only if the first mode is devdax
+#
+# If create the devdax namespace, we need someother space
+# from the same region to create a fsdax namespace.
+# For we need to store the pending_batch data and immutable configs
+# on fsdax mode.
+if [ "$type" == "devdax" ];then
+  device_name=$(ndctl create-namespace --region="$region_name" \
+                --size="$FSDAX_NAMESPACE_SIZE" --mode=fsdax | jq -r '.blockdev')
+  error_record_and_exit $? "ndctl create-namespace ... create the fsdax namespace"
+
+  mkfs.xfs -f /dev/"$device_name" | \
+      error_record_and_exit $? "mkfs.xfs -f /dev/'$device_name'"
+  mkdir -p /mnt/kvdk-pmem-meta | error_record_and_exit $? "mkdir "
```

Comment:
maybe declare a variable for /mnt/kvdk-pmem-meta?

Reply:
done